### ISSUE TITLE
ConstraintError: UNIQUE constraint failed: Torrent.infohash

### ISSUE THREAD

**Issue by synctext** [Tribler/tribler#1118]
When running V6.4.0 release for almost 3 weeks on my Ubuntu box.

```
ERROR   1418973202.49       community:617   Could not remove relay 1111594445 no activity
ERROR   1418973207.49       community:617   Could not remove relay 1730856164 no activity
ERROR   1418973212.49       community:617   Could not remove relay 3902713747 too old
ERROR   1418973227.95   sqlitecachedb:297   ===Twisted===
SQL Type: <type 'str'>
-----
INSERT INTO Torrent (infohash, status_id) VALUES (?, ?)
-----
[('6D0BMuwE76XMyKZC1IyEra8nogI=', 0), ('5S5qClXrZ/dR2YqoKgPvEINnFVo=', 0), ('390bcCZnD5eP9vrbx/McHGm4MdE=', 0), ('4sfk6UCKNWg3S1awUOE/udWzwNg=', 0), ('cz0VlqVu2Rj4ZEUYSbuoKBuq2Ms=', 0), ('DEApRbluuh1R6svB
l383izlAsGw=', 0)]
======
Traceback (most recent call last):
  File "/usr/share/tribler/Tribler/Core/CacheDB/sqlitecachedb.py", line 290, in executemany
    result = cur.executemany(sql, args)
  File "src/cursor.c", line 231, in resetcursor
ConstraintError: ConstraintError: UNIQUE constraint failed: Torrent.infohash
ERROR   1418973227.95        dispersy:1725  exception during handle_callback for dispersy-dynamic-settings
Traceback (most recent call last):
  File "/usr/share/tribler/Tribler/dispersy/dispersy.py", line 1720, in _update
    messages[0].handle_callback(messages)
  File "/usr/share/tribler/Tribler/community/channel/community.py", line 984, in on_dynamic_settings
    Community.on_dynamic_settings(self, *args, **kwargs)
  File "/usr/share/tribler/Tribler/dispersy/community.py", line 3604, in on_dynamic_settings
    meta.handle_callback(redo)
  File "/usr/share/tribler/Tribler/community/channel/community.py", line 357, in _disp_on_torrent
    self._channelcast_db.on_torrents_from_dispersy(torrentlist)
  File "/usr/share/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 1950, in on_torrents_from_dispersy
    torrent_ids, inserted = self.torrent_db.addOrGetTorrentIDSReturn(infohashes)
  File "/usr/share/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 619, in addOrGetTorrentIDSReturn
    self._db.executemany(sql, [(bin2str(infohash), status_id) for infohash in to_be_inserted])
  File "/usr/share/tribler/Tribler/dispersy/util.py", line 43, in helper
    return func(*args, **kargs)
  File "/usr/share/tribler/Tribler/Core/CacheDB/sqlitecachedb.py", line 298, in executemany
    raise msg
ConstraintError: ConstraintError: UNIQUE constraint failed: Torrent.infohash
ERROR   1418973312.57       community:617   Could not remove relay 960735092 no activity
ERROR   1418973342.58       community:617   Could not remove relay 2312011605 no activity
ERROR   1418973348.91       community:853   Don't know how to decrypt incoming message for circuit_id 1574842095
ERROR   1418973386.59       community:1046  Got destroy from {68.119.76.64:7759} for circuit 1956557363
ERROR   1418973386.59       community:1052  Got an exit socket 1956557363 ('68.119.76.64', 7759)
ERROR   1418973407.58       community:617   Could not remove relay 1465812917 no activity
ERROR   1418973427.58       community:617   Could not remove relay 2892998047 no activity
```

**Comment by NielsZeilemaker** [comment]
We've seen this one before, but I'm more amazed by all the "Could not remove relay ..." messages

**Comment by whirm** [comment]
@synctext do you still have the database that triggered this stacktrace? I want to take a look at the torrents table.

**Comment by LipuFei** [comment]
This one should be fixed through #1427.
